File: October/21-10-InsertDeleteGetRandom.py
import random
from secrets import choice

class RandomizedSet:

    def __init__(self):
        self.set = []

    def insert(self, val: int) -> bool:
        if val not in self.set:
            self.set.append(val)
            return True
        return False
        

    def remove(self, val: int) -> bool:
        if val in self.set:
            self.set.remove(val)
            return True
        return False
        

    def getRandom(self) -> int:
        # return random.choice(self.set) -> Solucao 1
        return choice(self.set) # Solucao 2
        
        
# Your RandomizedSet object will be instantiated and called as such:
# obj = RandomizedSet()
# param_1 = obj.insert(val)
# param_2 = obj.remove(val)
# param_3 = obj.getRandom()

File: 2021/October/21-10-InsertDeleteGetRandom.js

var RandomizedSet = function() {
    this.set = []
};

function get_random (list) {
    return list[Math.floor((Math.random()*list.length))];
}

/** 
  * @param {number} val
  * @return {boolean}
  */
RandomizedSet.prototype.insert = function(val) {
    if (!this.set.includes(val)) {
        this.set.push(val)
        return true
    }
    return false
};

/** 
  * @param {number} val
  * @return {boolean}
  */
RandomizedSet.prototype.remove = function(val) {
    if (this.set.includes(val)) {
        const indexVal = this.set.indexOf(val)
        this.set.splice(indexVal,1)
        return true
    }
    return false
};

/**
  * @return {number}
  */
RandomizedSet.prototype.getRandom = function() {
    return get_random(this.set)
};

/** 
  * Your RandomizedSet object will be instantiated and called as such:
  * var obj = new RandomizedSet()
  * var param_1 = obj.insert(val)
  * var param_2 = obj.remove(val)
  * var param_3 = obj.getRandom()
  */